adapters/Vungle/VungleAdapter/GADMAdapterVungleRouter.m

```diff
@@ -229,13 +230,13 @@ - (void)removeDelegate:(nonnull id<GADMAdapterVungleDelegate>)delegate {
   } else if ([delegate respondsToSelector:@selector(isBannerAd)] && [delegate isBannerAd]) {
     @synchronized(_bannerDelegates) {
       if (delegate && (delegate == [_bannerDelegates objectForKey:delegate.bannerRequest])) {
-        if ([delegate.bannerRequest isEqualToBannerRequest:_bannerRequest]) {
-          if (!_isBannerPresenting && !delegate.isRequestingBannerAdForRefresh) {
-            _bannerRequest = nil;
-          }
-        }
         GADMAdapterVungleMapTableRemoveObjectForKey(_bannerDelegates, delegate.bannerRequest);
       }
+      if ([delegate.bannerRequest isEqualToBannerRequest:_bannerRequest] || ([delegate.bannerRequest.placementID isEqualToString:_bannerRequest.placementID] && !delegate.bannerRequest.uniquePubRequestID && !_bannerRequest.uniquePubRequestID)) {
```

Comment:
@YueVungle I'm not sure why this needs to change, since in other places where we check if we can load an ad again, we still only rely on [delegate.bannerRequest isEqualToBannerRequest:_bannerRequest]. Having said that, I'm not sure what the side effects are so I won't press on this. This looks good to me and we'll merge pending final testing from @joshuajplagonera

Reply:
@ericleich  Hi Eric, I making this change is for the scenario that when the refresh Banner request is failed to load ad and not added to _bannerDelegates, then users close this Banner view. The adapter should need to set _bannerRequest to nil to let the users can load another Banner ad with different GADMAdapterVungleBannerRequest. Thanks a lot. ^_^